tree_unique_args/src/lib.rs

```diff
@@ -44,9 +50,57 @@ pub enum Expr {
     Let(Id, Box<Expr>, Box<Expr>),
     Arg(Id),
     Function(Id, Box<Expr>),
+    /// A list of functions, with the first
+    /// being the main function.
+    Program(Vec<Expr>),
```

Comment:
A `Program` should not be an `Expr` variant.

Reply:
I've made `Program` an `Expr` for the same reason `Function` is an `Expr`.
In this encoding, we've made everything an `Expr` for convenience.

Reply:
Function is not an Expr, that was a mistake that I made.